pkg/cmd/checkendpoints/controller/metrics.go

```diff
@@ -0,0 +1,70 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoints/trace"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+var (
+	registerMetrics sync.Once
+
+	endpointCheckCounter   *metrics.CounterVec
+	tcpConnectLatencyGauge *metrics.GaugeVec
+	dnsResolveLatencyGauge *metrics.GaugeVec
+)
+
+func RegisterMetrics(prefix string) {
+	registerMetrics.Do(func() {
+		endpointCheckCounter = metrics.NewCounterVec(&metrics.CounterOpts{
+			Name: prefix + "endpoint_check_count",
+			Help: "Report status of endpoint checks for each pod over time.",
+		}, []string{"endpoint", "tcpConnect", "dnsResolve"})
```

Comment:
something about status codes too?

Reply:
> something about status codes too?

or is that a different metric.

Reply:
No status codes as it's simply a TCP connection. Would you like to see the kind of error somehow?

Reply:
> No status codes as it's simply a TCP connection. Would you like to see the kind of error somehow?

yes.  Not all failure modes are the same.  connection refused, no route to host, dial error, timeout, they alll mean differen things

Reply:
I would still like timeout, connection refused, no route to host to have separate counts so that we can see how they change over time.